packages/block-editor/src/components/inserter/hooks/use-clipboard-block.native.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { rawHandler, store as blocksStore } from '@wordpress/blocks';
+import { getClipboard } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../../../store';
+
+export default function useClipboardBlock( destinationRootClientId ) {
```

Comment:
The logic for this hook has been extracted from the [`getItems` function of the inserter menu](https://github.com/WordPress/gutenberg/blob/ae7ab4528ed8afd17334ea054249af3a666e50ad/packages/block-editor/src/components/inserter/menu.native.js#L152-L173).

Reply:
Since this is for mobile-only, what do you think if we put it under the same `components/src/mobile/clipboard` folder? I like that this was extracted 👍

Reply:
Oh good point, I'll put it there since it could be used in other places, thanks!

Reply:
One thing I'm a bit concerned about regarding moving it to `packages/components` is that we would introduce a dependency into `components` package to `block-editor` which might not a good idea. I'm not sure how we usually deal with cross packages dependencies, wdyt?

Reply:
Oh! You're right, I think we can leave it here 😃